### PR TITLE
Python3.5 support

### PR DESCRIPTION
**Pull request by suquark**
This is the version that only contains python3.5 support & tests.

### CODE REVIEW

**Comment by suquark** [comment]
@pitrou Ready for review. 

The ARM64 CI errors seem not related to this PR.

cc @pcmoritz

**Comment by pitrou** [comment]
There are conflicts now, can you recreate this PR on git master?

**Comment by suquark** [comment]
sure

**Comment by pcmoritz** [comment]
@suquark Thanks for the update, this is great! Can you also update the readme to say `It should work with Python 3.5, 3.6 and 3.7.`?

**Comment by suquark** [comment]
I have updated the README & docs

**Comment by pcmoritz** [comment]
Great thanks! Let us know what you think @pitrou :)

File: patch [hunk 1]
```diff
@@ -181,7 +183,46 @@
...
  import struct
  import sys
-@@ -32,7 +30,8 @@
+@@ -23,6 +21,38 @@
+ 
+ try:
+     import numpy as np
++
++    # This function is used for numpy serialization with pickle5 in Python3.5.
++    # It can be removed once numpy enables Pickle Protocol 5 for Python3.5.
```
Comment (pitrou) [inline patch:192]: This doesn't sound necessary. Just don't test Numpy with Python 3.5. Numpy should run their own tests anyway.

File: BACKPORT-NOTES.txt [hunk 2]
```diff
@@ -12,4 +12,6 @@ then:
 
 * Same for 3.7
```
Comment (pitrou) [inline BACKPORT-NOTES.txt:13]: And 3.5, right?

File: .travis.yml [hunk 3]
```diff
@@ -13,6 +13,11 @@ env:
 
 jobs:
   include:
+  - python: "3.5"
+    dist: trusty
+    arch: amd64
+  - python: "3.5"
+    arch: amd64
```
Comment (pitrou) [inline .travis.yml:20]: Can you add Python 3.5 entries to the AppVeyor configuration as well?

**Comment by suquark** [comment]
I have updated the code as requested.

**Comment by pitrou** [comment]
Thank you @suquark. This looks good to me now.

**Comment by pcmoritz** [comment]
@pitrou Thanks for merging this! Can we also make a new pickle5 release with these changes? Also, would it be possible to create wheels for pickle5?

**Comment by pitrou** [comment]
To create wheels, someone has to provide the scaffolding (I suppose build scripts or CI infrastructure) and then tell me how to use it.

**Pull request merged**